### DISCUSSION TITLE
Deploy ejbca.ear error

### DISCUSSION THREAD

**Discussion by StocktonC** [Keyfactor/ejbca-ce#526] [Q&A]
hi all,
i'm trying to use latest Wildfly 31 (full version) + EJBCA v8.2.0.1 (ejbca-ce-EJBCA_8_2_0_1_20231206.zip)
When i start deploy the ejbca.ear, it return the following error:

"{
    \"WFLYCTL0080: Failed services\" => {
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"adminweb.war\\\".INSTALL\" => \"WFLYSRV0153: Failed to process phase INSTALL of subdeployment \\\"adminweb.war\\\" of deployment \\\"ejbca.ear\\\"
    Caused by: org.jboss.as.server.deployment.DeploymentUnitProcessingException: WFLYEJB0405: No Jakarta Enterprise Beans found with interface of type 'org.ejbca.core.ejb.EjbBridgeSessionLocal' and name 'EjbBridgeSessionBean' for binding java:module/env/EjbBridgeSession\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"healthcheck.war\\\".INSTALL\" => \"WFLYSRV0153: Failed to process phase INSTALL of subdeployment \\\"healthcheck.war\\\" of deployment \\\"ejbca.ear\\\"
    Caused by: org.jboss.as.server.deployment.DeploymentUnitProcessingException: WFLYEJB0405: No Jakarta Enterprise Beans found with interface of type 'org.ejbca.core.ejb.EjbBridgeSessionLocal' and name 'EjbBridgeSessionBean' for binding java:module/env/EjbBridgeSession\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"scep.war\\\".INSTALL\" => \"WFLYSRV0153: Failed to process phase INSTALL of subdeployment \\\"scep.war\\\" of deployment \\\"ejbca.ear\\\"
    Caused by: org.jboss.as.server.deployment.DeploymentUnitProcessingException: WFLYEJB0405: No Jakarta Enterprise Beans found with interface of type 'org.ejbca.core.ejb.EjbBridgeSessionLocal' and name 'EjbBridgeSessionBean' for binding java:module/env/EjbBridgeSession\"
    },
    \"WFLYCTL0412: Required services that are not installed:\" => [
        \"jboss.naming.context.java.EjbcaDS\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"adminweb.war\\\".jndiDependencyService\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"scep.war\\\".jndiDependencyService\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"healthcheck.war\\\".deploymentCompleteService\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"adminweb.war\\\".deploymentCompleteService\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"adminweb.war\\\".beanmanager\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"scep.war\\\".beanmanager\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"scep.war\\\".deploymentCompleteService\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"healthcheck.war\\\".beanmanager\",
        \"jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"healthcheck.war\\\".jndiDependencyService\"
    ],
    \"WFLYCTL0180: Services with missing/unavailable dependencies\" => [
        \"service jboss.deployment.unit.\\\"ejbca.ear\\\".deploymentCompleteService is missing [jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"scep.war\\\".deploymentCompleteService, jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"healthcheck.war\\\".deploymentCompleteService, jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"adminweb.war\\\".deploymentCompleteService]\",
        \"service jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"healthcheck.war\\\".batch.artifact.factory is missing [jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"healthcheck.war\\\".beanmanager]\",
        \"service jboss.persistenceunit.\\\"ejbca.ear#ejbca\\\".__FIRST_PHASE__ is missing [jboss.naming.context.java.EjbcaDS]\",
        \"service jboss.persistenceunit.\\\"ejbca.ear#ejbca\\\" is missing [jboss.naming.context.java.EjbcaDS]\",
        \"service jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"adminweb.war\\\".weld.weldClassIntrospector is missing [jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"adminweb.war\\\".beanmanager]\",
        \"service jboss.deployment.unit.\\\"ejbca.ear\\\".WeldStartService is missing [jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"healthcheck.war\\\".jndiDependencyService, jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"adminweb.war\\\".jndiDependencyService, jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"scep.war\\\".jndiDependencyService]\",
        \"service jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"adminweb.war\\\".batch.artifact.factory is missing [jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"adminweb.war\\\".beanmanager]\",
        \"service jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"scep.war\\\".batch.artifact.factory is missing [jboss.deployment.subunit.\\\"ejbca.ear\\\".\\\"scep.war\\\".beanmanager]\"
    ]
}"

Platform:
OS: RockyLinux8
DB: MySQL 8.0.36
Openjdk 11
Wildfly 31 (full version)
EJBCA v8.2.0.1 (ejbca-ce-EJBCA_8_2_0_1_20231206.zip)
Ant 1.10.14

Which steps did i miss? Thanks.

**Comment by primetomas** [comment]
It's not spelled out explicitly here:
https://doc.primekey.com/ejbca/ejbca-installation/application-servers

But it says JEE7 compatible. WildFly >26 moved to JakartaEE which is not backwards compatible. (the first time Java EE broke backwards compatibility ever). This means that it's a hard break for an EJBCA version that works with WildFly26 and current JBoss EAP versions and one that works with WildFly >26. 
Currently there is no supported version of JBoss EAP that is updated for Jakarta EE, but we have a roadmap for EJBCA to make this break once there is. Then you can start using WildFly >26, until then we are stuck due to how Jakarta EE made this update process.
